huaweicloud/resource_huaweicloud_scm_certificate_v3.go

```diff
@@ -0,0 +1,408 @@
+			},
+			"issue_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"not_before": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"not_after": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"certificate_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"validation_method": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"validity_period": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"wildcard_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"authentifications": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"record_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"record_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"record_value": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"domain": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					}},
+			},
+		},
+	}
+}
+
+func resourceScmCertificateV3Create(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	elbClient, err := config.ScmV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating HuaweiCloud SCM client: %s", err)
+	}
+
+	importOpts := certificates.ImportOpts{
+		Name:             d.Get("name").(string),
+		PrivateKey:       d.Get("private_key").(string),
+		Certificate:      d.Get("certificate").(string),
+		CertificateChain: d.Get("certificate_chain").(string),
+	}
+
+	c, err := certificates.Import(elbClient, importOpts).Extract()
+	if err != nil {
+		return fmt.Errorf("Error importing certificate: %s", err)
+	}
+	// If all has been successful, set the ID on the resource
+	d.SetId(c.CertificateId)
+	log.Printf("[DEBUG] Imported certificate %s: %#v", d.Id(), importOpts)
+
+	pushCert := d.Get("push_certificate").([]interface{})
+	for _, pushInfo := range pushCert {
+		targetService := pushInfo.(map[string]interface{})["target_service"].(string)
+		targetProjectArr := pushInfo.(map[string]interface{})["target_project"].([]interface{})
+
+		for _, targetProject := range targetProjectArr {
+			pushOpts := certificates.PushOpts{
+				TargetProject: targetProject.(string),
+				TargetService: targetService,
+			}
+			log.Printf("[DEBUG] Push certificate to services. %#v", pushOpts)
+			err := doPushCertificateToService(d.Id(), pushOpts, elbClient)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return resourceScmCertificateV3Read(d, meta)
+}
+
+func resourceScmCertificateV3Update(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*config.Config)
+	elbClient, err := config.ScmV3Client(GetRegion(d, config))
+
+	oldVal, newVal := d.GetChange("push_certificate")
```

Comment:
suggest to define it as schema.Set, then we can use `Difference` method to get the new target

Reply:
some logic has been updated.